### DISCUSSION TITLE
SMTP on local hosted server

### DISCUSSION THREAD

**Discussion by jarvismeierbrt6hc** [immense/Remotely#341] [Q&A]
I have a Barracuda Spam Firewall that is currently configured with no auth and no TLS for SMTP.  Any way to get SMTP working with these requirements?

Alternately I have an exchange hybrid environment with a local mailbox it can use however the SMTP over TLS certificate is self signed and it's being rejected.  How does one import the Root and Issuing CA's so they can be used by remotely?  

This is running on an ubuntu 20.04 server.

**Comment by bitbound** [comment]
For the first scenario, I think I'd have to add some configuration options.  But I don't have a similar environment to test against, and my time's stretched a little thin currently.

I've never added a self-signed cert to Ubuntu, and I'm not sure if ASP.NET Core will automatically trust it.  But it's worth a shot.  Here's what I found:

https://askubuntu.com/questions/645818/how-to-install-certificates-for-command-line

I'd also disable "Check for Certificate Revocation" in Remotely Server Config.

Afterward, I'd restart the Remotely service, just to be sure: `sudo systemctl restart remotely` 

Let me know if that works.  :)

**Reply by jarvismeierbrt6hc** [reply]
I tried and it doesn't work.  ASP.net core doesn't seem to use the global certificates for email.  If you are using mailkit i found this:
 client.ServerCertificateValidationCallback

Using this one should be able to bypass certificate verification.

I also found this from someone else who was having a similar issue.  This is what he came back with:

---------------------------------------------------
I finally figured out why this stopped working on Linux. The netcore3.1 framework does not read from the OS's ca-certificates folder, but from

~/.dotnet/corefx/cryptography/x509stores/ca

Therefore the app did not trust ANY certificates. So I downloaded Google's sample PEM file from this page, converted it into a P7B file to make sure the entire certificate chain remains intact

openssl crl2pkcs7 -nocrl -certfile google-roots.pem -out google-roots.p7b

and added the entire chain to the framework's certificate store during app startup

using (var store = new X509Store(StoreName.CertificateAuthority, StoreLocation.CurrentUser))
{
    store.Open(OpenFlags.ReadWrite);

    options.CertificateChainFiles.ToList().ForEach(file =>
    {
        var collection = new X509Certificate2Collection();
        collection.Import(file);
        store.AddRange(collection);
    });
}
---------------------------------------------------

I'm not sure what user Remotely runs as but if it's www-data the cert store it should read from is /var/www/.dotnet/corefx/cryptography/x509stores/ca
